Modify rule S2761: LaYC multiple unary operators (#2180)
SonarSource/rspec: rules/S2761/php/rule.adoc

== Why is this an issue?

> Calling the ``++!++`` or ``++~++`` prefix operator twice is usually a typo. The second invocation invalidates the first one:

Reviewer: ```suggestion
The repetition of a prefix operator (``++!++``, or ``++~++``) is usually a typo. The second operator invalidates the first one:
```

> Calling the ``++!++`` or ``++~++`` prefix operator twice does nothing: the second invocation undoes the first. Such mistakes are typically typos. Either this is a bug if the operator was meant to be called once, or misleading if done on purpose.

Reviewer: ```suggestion
Calling the ``++!++`` or ``++~++`` prefix operator twice is usually a typo. The second invocation invalidates the first one:
```

[source,php]
----
$a = false;
$b = ~~$a; // Noncompliant: Same as $a
----

> Calling ``++!++`` twice is sometimes done instead of using the dedicated `boolval()` function. However, the latter increases the code readability and should be preferred.

Reviewer: ```suggestion
While calling ``++!++`` twice is equivalent to calling the `boolval()` function, the latter increases the code readability, so it should be preferred.
```
